game/menu/gamemanager.gd

```diff
@@ -39,37 +190,36 @@ func _build_menu():
 	_main.show()
 
 	#making the buttons
-	for game in _games:
+	for game_id in GAMES.keys():
 		var display = _preview_scene.instance()
-		display.setup(game)
+		display.setup(GAMES[game_id])
 		display.connect("pressed", self, "load_game")
 		_grid.add_child(display)
+		GAME_DISPLAYS[game_id] = display
 
 
 # go through every folder inside res://games/ and try to load the game.cfg into _games
 func _find_games():
-	_games.clear()
+	GAMES.clear()
 	var dir = Directory.new()
 	if dir.open("res://games") == OK:
 		dir.list_dir_begin(true)
 		var file_name = dir.get_next()
 		while file_name != "":
 			if dir.current_is_dir():
-				var game_path = "res://games/" + file_name + "/game.cfg"
-
-				var err := _load_game_cfg_file(game_path)
+				var err := _load_game_cfg_file(file_name)
 				if err != OK:
 					prints("Error loading game cfg:", err)
 
 			file_name = dir.get_next()
 
 
 # load a config file into _games
-func _load_game_cfg_file(path: String) -> int:
-	var f := ConfigFile.new()
-	var err := f.load(path)
+func _load_game_cfg_file(folder_name: String) -> int:
+	var config := ConfigFile.new()
+	var err := config.load("res://games/" + folder_name + "/game.cfg")
 	if err != OK:
 		return err
-	f.set_meta("folder_path", path.get_base_dir() + "/")
```

Comment:
The path would just be duplicate information

Reply:
Disagree, especially #37 needs it for visualization.

Reply:
That PR isn't about visuals. So why would it need that for visualization?
Anyway, maybe we should include that path anyway just for convenience?
`cfg.get_meta("folder_path")` is easier than
`"res://games/%s/" % cfg.get_meta("folder_name")`.

Reply:
We could have a method that converts the folder name to the folder path. That would be nicer, with an optional file_name that can be added to the end

Reply:
I don't see how removing helpful information for error handling is a good idea.
I think we could live without explicit references to a config object and use getters and setters from our own class.
Then implementing folder_path could just return a templated string.
Otherwise removing that setting from the store seems counter productive.

Especially error messages a'la "Could not find res://games/meow/config.cfg are you sure it exists?" are better for developer experience than falling back to using debugger errors. Ignoring the obvious issue of warnings being issued for missing error handling.

Then again i could also live with a more capsuled approach to all of this, instead of relying on Strings inside a config object.

Reply:
> I don't see how removing helpful information for error handling is a good idea.

It is not removing any information. It just makes it a bit harder to access the information